src/libraries/System.Private.CoreLib/src/System/Resources/ResourceReader.cs

```diff
@@ -380,10 +380,7 @@ private unsafe string AllocateStringForNameIndex(int index, out int dataOffset)
                     if (_ums.Position > _ums.Length - byteLen)
                         throw new BadImageFormatException(SR.Format(SR.BadImageFormat_ResourcesIndexTooLong, index));
 
-                    string? s = null;
-                    char* charPtr = (char*)_ums.PositionPointer;
-
-                    s = new string(charPtr, 0, byteLen / 2);
+                    string s = Encoding.Unicode.GetString(_ums.PositionPointer, byteLen);
```

Comment:
What is the perf impact of this change?  Previously it was a memcpy; now it's going through Encoding.Unicode.

Reply:
In this case, the change looked like a potential correctness (or at least consistency) fix even on little-endian to me.   This routine has two code paths, with a special case where the resource file is provided as an UnmanagedMemoryStream.  In all other cases, the existing code already used Encoding.Unicode.GetString, only the UMS case used the direct copy.

These are semantically different e.g. in the case when surrogate characters are present.  It seems strange to have this semantic difference triggered by simply whether the input stream is a UMS or not.

If this different behavior is indeed intentional, I guess I can instead add a simple byte-swapping loop in the big-endian case.

(As to perf impact, I don't know what would be a good benchmark for that.  Looking at the Encoding.Unicode source code, there is a fast path for the case where no surrogates are present, but that is of course still slower than just a memcpy.)